transport/http/handler.go

```diff
@@ -59,6 +59,11 @@ func (h handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	if err == context.DeadlineExceeded {
+		http.Error(w, "remote timeout", http.StatusGatewayTimeout)
```

Comment:
there is no local vs remote version of the timeout error in tchannel. So I decided to not add the distinction in yarpc. So really, there is 3 kind of timeout error: local, remote and "local_or_remote_i_cant_tell". With HTTP the distinction is clear (the http transport returns deadlines errors locally, and we have the 504 for a remote returning a timeout). TChannel has a higher abstraction, returning the same timeout error code for local timeout and remote timeout.

The distinction is mostly for human consumption and for a couple tests in crossdock. I decided to use a single timeout error, with an msg containing extra details (and I am all for adding more details as you proposed).